### PR TITLE
Switch to using helm3 for bootstrap

### LINKED ISSUE #184: Convert all kubectl create commands to kubectl apply
## Task

Convert all kubectl create commands to kubectl apply

Something like this may be better:

```
kubectl create etc etc etc -o yaml --dry-run | kubectl apply -f
```

### PR DESCRIPTION
**Pull request by alexellis**
Switch to using helm3 for bootstrap

There's an ongoing issue which has been patched in helm 3 but
which is not yet patched in helm 2. We also have a backlog item
to move to the secure by default helm 3.

This change also downloads helm 3 and uses a newer version of
go-execute which has a fix for setting the $PATH correctly to
access binaries we have downloaded.

* Use faas-cli to get kubeseal
* Apply secrets in re-runnable way in bash scripts to fix #184
* Update for local usage of helm3 path
* Also patches minio to use helm3-style argument for install
* PATH is now set for the process before invoking execTask so that
the helm binary can be found.
* Use rollout status for sealedsecrets - this is quicker and uses
less forking than a loop with an outer check.
* Update deps
* Use non-root user for Dockerfile
* Add note for k3d without Traefik
* Use helm3 to install cert-manager
* Add k3sup to Gopkg.toml
* Remove reset scripts - the reset scripts became a source of
confusion for the community and due to the amount of software
installed, it's hard to truely reset a cluster and purge all
API extensions / charts.

Fixes: #178
Fixes: #181
Fixes: #180
Closes: #183
Fixes: #184

Tested across two main scenarios:

* No TLS, no OAuth running inside k3d with inlets-operator for ingress
* DigitalOcean Kubernetes Service DOKS with full TLS and OAuth enabled

Signed-off-by: Alex Ellis (OpenFaaS Ltd) <alexellis2@gmail.com>

## Notes

* The reset scripts have been removed due to confusion caused and difficulty tracing and effectively removing all side-effects of helm charts like cert-manager.

* Tiller is removed

* `--wait` is used in some of the helm chart install scripts

* The Helm3 binary is downloaded automatically and used from a local copy

### CODE REVIEW

**Comment by alexellis** [comment]
Depends on https://github.com/alexellis/k3sup/issues/163

**Comment by alexellis** [comment]
@Waterdrips PTAL if you have time for a review and test.

I've gone e2e with this without TLS and without OAuth2 as follows.

**Comment by Waterdrips** [comment]
@alexellis - removing the reset scripts has broken CI
```
./hack/integration-test.sh: line 5: ./scripts/reset-kind.sh: No such file or directory
```

File: cmd/apply.go [hunk 1]
```diff
@@ -317,42 +350,49 @@ func process(plan types.Plan, prefs InstallPreferences) error {
...
-func helmRepoUpdate() error {
-	log.Println("Updating helm repos")
+func buildPath(additionalPaths []string) string {
+
+	pathVal := "PATH="
+	for _, p := range additionalPaths {
+		pathVal = pathVal + p + ":"
+	}
+
+	// pathVal = strings.TrimRight(pathVal, ":")
```
Comment (Waterdrips) [inline cmd/apply.go]: if you are committing again it would be nice to get rid of this.

Comment (alexellis) [reply cmd/apply.go]: Thanks I'll fix that.

**Pull request merged**